packages/next-auth/src/providers/boxyhq-saml.ts

```diff
@@ -0,0 +1,37 @@
+import type { OAuthConfig, OAuthUserConfig } from "."
+
+export interface SAMLJacksonProfile {
+  id: string;
+  email?: string;
+  firstName?: string | null;
+  lastName?: string | null;
+}
+
+export default function SAMLJackson<
+  P extends Record<string, any> = SAMLJacksonProfile
+>(options: OAuthUserConfig<P>): OAuthConfig<P> {
+  return {
+    id: "boxyhq-saml",
+    name: "BoxyHQ SAML Jackson",
+    type: "oauth",
+    version: "2.0",
+    checks: ["pkce", "state"],
+    authorization: {
+      url: `${options.issuer}/api/oauth/authorize`,
+      params: {
+        provider: "saml",
+      },
+    },
+    token: `${options.issuer}/api/oauth/token`,
+    userinfo: `${options.issuer}/api/oauth/userinfo`,
+    profile(profile) {
+      return {
+        id: profile.id,
+        email: profile.email || "",
```

Comment:
```suggestion
        email: profile.email,
```

You mentioned this being guaranteed: https://github.com/nextauthjs/next-auth/pull/3782#discussion_r799064357